cmake/modules/CheckCompiler.cmake

```diff
@@ -192,8 +193,19 @@ check_cxx_source_compiles(
 int main() {}
 " GLIBCXX_USE_CXX11_ABI)
 
+
 #---Print the final compiler flags--------------------------------------------------------------------
 message(STATUS "ROOT Platform: ${ROOT_PLATFORM}")
 message(STATUS "ROOT Architecture: ${ROOT_ARCHITECTURE}")
 message(STATUS "Build Type: ${CMAKE_BUILD_TYPE}")
-message(STATUS "Compiler Flags: ${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_${uppercase_CMAKE_BUILD_TYPE}}")
+
+string(TOUPPER ${CMAKE_BUILD_TYPE} upperCaseBuildType)
+message(STATUS "Compiler Flags (automatically from 'CMAKE_BUILD_TYPE'): ${CMAKE_CXX_FLAGS_${upperCaseBuildType}}")
+
+#---Fix order of compiler flags: user-supplied flags are overridden otherwise----------------------
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS_${upperCaseBuildType}} ${CMAKE_CXX_FLAGS}")
+unset(CMAKE_CXX_FLAGS_${upperCaseBuildType})
+
+message(STATUS "Final compiler Flags: ${CMAKE_CXX_FLAGS}")
```

Comment:
Can we leave just the final compiler flags (the other looks like a debug trace)? And why is "Flags" caps?

Reply:
As you wish ...
- Previously, flags was caps. It's a copy-and-paste error.
- The printing of the automatic compiler flags helped me understand why ROOT seems to be doing strange things such as putting flags like:
`-g -O2 -O3 -march=native`
If I had been a user setting `-DCMAKE_CXX_FLAGS="-O3 -march=native"`, I would wonder why the heck somebody puts -g -O2 in front.
(Actually, I would have wondered why somebody puts it *after* my flags, but that's what we are fixing at the moment)

Reply:
I disagree, we also have no tracing info in our C++ code. And you can use the same arguments for sprinkling `printf`s throughout ROOT's C++ code, which shows that "it allows me to understand what's happening" isn't the main purpose of that code :-)